### PR TITLE
[MM-35922] Fixed a couple style issues with the sidebar category headers on drag

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Fixed two issues with styles on the LHS when dragging categories or over categories:
- When dragging a channel over a category header, or an autosorted category, the border was on the wrong element. I've added the border in the correct places.
- When dragging a category, the `box-shadow` was also on the wrong element, so that has been adjusted as well.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35922

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]

**Comment by amyblais** [comment]
@devinbinnie Kind reminder to take a look at Jelena's comment ^

File: sass/layout/_sidebar-left.scss [hunk 1]
```diff
@@ -612,15 +612,21 @@
...
+            }
+        }
     }
 
     .SidebarChannelGroup:not(.autoSortedCategory) .draggingOver .SidebarChannelGroupHeader:hover {
         box-shadow: inset -1px 0 0 rgba(var(--sidebar-text-rgb), 0.60), inset 0 -1px 0 rgba(var(--sidebar-text-rgb), 0.6), inset 1px 0 0 rgba(var(--sidebar-text-rgb), 0.6), inset 0 1px 0 rgba(var(--sidebar-text-rgb), 0.6);
         border-radius: 4px;
-
-        .SidebarChannelGroupHeader_groupButton {
-            background-color: inherit;
```
Comment (devinbinnie) [inline sass/layout/_sidebar-left.scss:622]: I'm not entirely sure why this was added, but removing it makes sure that the issues with the border go away.

**Comment by devinbinnie** [comment]
> Thanks @devinbinnie
> Looks good, one small issue - dragging channel over autosorted category breaks the border a bit on the category name, right side, vertical scrollbar area. See image.
> Screen Shot 2021-05-26 at 10 15 43 AM

@jgilliam17 Have another look when you have a second.

**Review by jgilliam17 — APPROVED** [review]
Thanks @devinbinnie 
Tested, looks good to merge.
- Verified sidebar category and headers border on channel drag & drop.
Ping me here again if a retest is needed after Harrison takes a look. Thanks.

**Review by michelengelen — APPROVED** [review]
LGTM!

**Comment by amyblais** [comment]
@michelengelen Would you be able to cherry-pick this to the ``cloud`` branch as Devin is off this week?

**Comment by michelengelen** [comment]
Hi @amyblais ... cherry-pick done with the this [commit](https://github.com/mattermost/mattermost-webapp/commit/b808e4385bdd524940dd96ce49337efe69040b0a)

**Pull request merged**